web-app/src/index.html

```diff
@@ -38,6 +38,28 @@
       }
     }
 
+    function apiGet(patternInfo) {
+      return fetch(SM.apiUrl + patternInfo.links.find(link => link.rel === 'self').href,
+        {
+          headers: this.getAuthHeaders()
+        });
+    }
+
+    function apiGetThumbnail(patternInfo) {
```

Comment:
Again, why would global context know about pattern info? This function does not belong here.